### PR TITLE
feat: enable access control for media source

### PR DESCRIPTION
**Pull request by gabrielfs7**
feat: enable access control for media source

https://oat-sa.atlassian.net/browse/ADF-271

- [ ] https://github.com/oat-sa/tao-core/pull/2908
- [ ] https://github.com/oat-sa/extension-tao-mediamanager/pull/269

### CODE REVIEW

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by voknil — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: models/classes/media/AssetTreeBuilderInterface.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+     *              "permissions": [
+     *                  "READ",
+     *                  "WRITE",
+     *              ],
+     *              "url": "https://taotesting.com/taoItems/ItemContent/files",
+     *          ]
+     *      ]
+     *   ]
+     */
+    public function build(DirectorySearchQuery $search): array;
```
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply models/classes/media/AssetTreeBuilderInterface.php:53]: I kept array, cause it was the previous signature.

File: models/classes/media/AssetTreeBuilderInterface.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace oat\taoItems\model\media;
+
+use oat\tao\model\media\mediaSource\DirectorySearchQuery;
+
+interface AssetTreeBuilderInterface
+{
+    /**
+     * @example Returns:
+     *  [
```
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply models/classes/media/AssetTreeBuilderInterface.php:31]: I think in this case, here is better place than README, cause anyone using this interface would be able to know the response just by using IDE right click / mouse over. 

I do not think they would go to README.md to read that. Btw, this is not the API response it is an internal class response.

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**